src/vuln_analysis/utils/source_code_git_loader.py

```diff
@@ -128,8 +129,13 @@ def load_repo(self):
             logger.info("Using existing Git repo at path: '%s' @ '%s'", self.repo_path, self.ref)
 
         # Reliable way to check out the ref using a shallow clone
-        repo.git.fetch("origin", self.ref, depth=1)
-        repo.git.checkout("FETCH_HEAD")
+        repo.git.fetch("origin", self.ref, "--depth=1", "--force")
+        tag_refspec = f"refs/tags/{self.ref}:refs/tags/{self.ref}"
+        try:
+            repo.git.fetch("origin", tag_refspec, "--force")
```

Comment:
@vbelouso In case  it will not fail ( if the ref is a tag), then it will fetch the commits tree with the full history of it, then need to do it shallow as well.
```suggestion
            repo.git.fetch("origin", tag_refspec, "--depth=1" , "--force")
```